### DISCUSSION TITLE
[MTR05] week8 hw2 Twitch API 串接無法發出 request

### DISCUSSION THREAD

**Discussion by zoeliuhy** [Lidemy/forum#141] [程式導師實驗計畫]
## Question：
串接 Twitch API 的部分，有依照老師的講解影片修改，但還是發不出 request ，打開 devtool 的 network 看，Request URL 是本地的資料夾(Request URL: file:///C:/Users/zoeliu/Documents/lidemy/mentor-program-5th-ZoeLiuhy/homeworks/week8/hw2/index.js)、Request header 則是出現: Provisional headers are shown ，但是一直找不出程式碼有哪裡出錯，因為已經卡關滿久，來這邊求助大家...

## 程式碼：
```
const clientId = 'cggmzqykplcvt5zqpl5aofaul8ljhp'
const api = 'https://api.twitch.tv/kraken'
const errorMessage = 'ERROR'

function responseTopGames(callback) {
  const request = new XMLHttpRequest()
  request.open('GET', api + '/games/top?limit=5', true)
  request.setRequestHeader('Client-ID', clientId)
  request.setRequestHeader('Accept', 'application/vnd.twitchtv.v5+json')
  
  request.onload = function() {
    if (request.status >= 200 && request.status < 400) {
      console.log(JSON.parse(request.response))
    } else {
      callback(errorMessage)
    }
  }
  request.onerror = function(e) {
    callback(errorMessage)
  }
  request.send()
}
```
## devtool 訊息：
![螢幕擷取畫面 2021-06-13 103028](https://user-images.githubusercontent.com/59194225/121793592-2a4cb280-cc33-11eb-91a5-ab29a7544163.png)

**Comment by allenliaooooooo** [comment]
建議可以在程式碼的前後各加上三個 ` 符號
而且至少要縮排
讓大家比較好閱讀

**Reply by zoeliuhy** [reply]
@rockyooooooo 感謝建議，下次要發問題之前我會好好調整一下排版的!!

**Comment by cwc329** [comment]
你看 error message，
這邊想要取得的資源是 `file:///C:/Users/zoeliu/Documents/lidemy/mentor-program-5th-ZoeLiuhy/homeworks/week8/hw2/index.js`，
這個應該是你想要取用的 JS 檔，
所以應該不是 twitch 那邊出問題，
而是你 JS 檔的路徑有問題。

**Reply by zoeliuhy** [reply]
@cwc329 我試了直接把資料放到html裡面的script裡面，也還是無法console出資料

**Comment by aszx87410** [comment]
看一下 console 有沒有任何錯誤，然後 devtool 的訊息用截圖的比較準

**Reply by zoeliuhy** [reply]
@aszx87410 console完全是空白的><

![螢幕擷取畫面 2021-06-13 103028](https://user-images.githubusercontent.com/59194225/121793517-6c292900-cc32-11eb-8a00-0842785951bb.png)

**Reply by aszx87410** [reply]
你可以仔細看一下左邊 name 那邊寫著 index.js，所以你截圖這個 request 是去拿 index.js 這個檔案，跟你的 API 一點關係都沒有
再加上從左邊看來，並沒有任何發送去 twitch 的 API
所以我猜你是忘記呼叫 function，`responseTopGames()`

**Reply by zoeliuhy** [reply]
@aszx87410 成功抓到資料了!!!!!! 感謝!!
